### PR TITLE
[MTPG] Enable for some tests in test_fsdp_misc

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102043

Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with @kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

### CODE REVIEW

**Comment by fduwjj** [comment]
Looks like some distributed tests failed?

File: test/distributed/fsdp/test_fsdp_misc.py [hunk 1]
```diff
@@ -184,46 +261,74 @@ def _check_equal(local, fsdp):
...
     @skip_if_lt_x_gpu(2)
-    @parametrize("use_second_layer", [True, False])
-    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD, None])
-    def test_fsdp_module_no_compute_grad(self, use_second_layer, sharding_strategy):
-        # When use_second_layer=True, b is involved in forward computation but does
-        # not receive grad in backward. Otherwise, b is not involved in forward
-        # computation.
-        class MyModel(nn.Module):
+    def test_fsdp_cpu_init_stays_on_cpu(self):
+        # Move me to MT test once warning logging and backward collective issue
```
Comment (awgu) [inline test/distributed/fsdp/test_fsdp_misc.py:265]: Should we file an issue to track?

**Review by awgu — APPROVED** [review]
Lint error looks real :/

**Review by awgu — APPROVED** [review]
Lint error looks real :/

**Pull request closed without merging**